Add Time-Averaged Field Diagnostics (#5285)
ECP-WarpX/WarpX: Docs/source/usage/parameters.rst

In-situ visualization
^^^^^^^^^^^^^^^^^^^^^

WarpX has five types of diagnostics:

> ``FullDiagnostics`` consist in dumps of fields and particles at given iterations,
> ``TimeAveragedDiagnostics`` only allow field data which they output after averaging over a period of time,
> ``BackTransformedDiagnostics`` are used when running a simulation in a boosted frame, to reconstruct output data to the lab frame,
> ``BoundaryScrapingDiagnostics`` are used to collect the particles that are absorbed at the boundary, throughout the simulation, and

Reviewer: Maybe an opportunity to change the names exposed here:
```suggestion
``Full`` diagnostics consist in dumps of fields and particles at given iterations,
``TimeAveraged`` diagnostics only allow field data, which they output after averaging over a period of time,
``BackTransformed`` diagnostics are used when running a simulation in a boosted frame, to reconstruct output data to the lab frame,
``BoundaryScraping`` diagnostics are used to collect the particles that are absorbed at the boundary, throughout the simulation, and
```
I think they do not match the actual input names, do they?

* ``warpx.mffile_nstreams`` (`int`) optional (default `4`)
    Limit the number of concurrent readers per file.

.. _running-cpp-parameters-diagnostics-timeavg:

Time-Averaged Diagnostics
^^^^^^^^^^^^^^^^^^^^^^^^^

> ``TimeAveraged`` diagnostics are a special type of ``FullDiagnostics`` that allows for the output of time-averaged field data.

Reviewer: ```suggestion
``TimeAveraged`` diagnostics are a special type of ``Full`` diagnostics that allows for the output of time-averaged field data.
```

This type of diagnostics can be created using ``<diag_name>.diag_type = TimeAveraged``.
We support only field data and related options from the list at `Full Diagnostics`_.

.. note::

>     As with ``FullDiagnostics``, ``TimeAveraged`` diagnostics output the initial **instantaneous** conditions of the selected fields on step 0 (unless more specific output intervals exclude output for step 0).

Reviewer: ```suggestion
    As with ``Full`` diagnostics, ``TimeAveraged`` diagnostics output the initial **instantaneous** conditions of the selected fields on step 0 (unless more specific output intervals exclude output for step 0).
```
